### PR TITLE
1e-3 check for is amount counter paid

### PR DESCRIPTION
**Pull request by pskd73**

### CODE REVIEW

File: pyluca/round_off.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+PRECISION_ZERO = 3
+
+
+def zeroed(amount: float):
```
Comment (rahulsekar) [inline pyluca/round_off.py:4]: is_zero()

Comment (rahulsekar) [reply pyluca/round_off.py:4]: return -1e-3 < amount < 1e-3

File: pyluca/aging.py [hunk 2]
```diff
@@ -26,8 +26,9 @@ def _pay_counters(positive_entries: List[PositiveEntry], amount: float, date: da
         return 0
     rem_amount, cur_idx = amount, 0
     while rem_amount > 0 and cur_idx < len(positive_entries):
-        _, rem_amount = positive_entries[cur_idx].counter.pay(rem_amount, date)
-        if positive_entries[cur_idx].counter.is_paid():
+        if not is_amount_counter_paid(positive_entries[cur_idx].counter):
```
Comment (pskd73) [inline pyluca/aging.py:29]: This is the culprit. This solves the problem

File: pyluca/amount_counter.py [hunk 3]
```diff
@@ -48,13 +41,14 @@ def pay(self, amount: float, date: datetime):
...
         return self.total_amount - self.paid_amount
 
-    def is_paid(self):
-        return abs(self.get_balance()) < TOLERANCE_FLOATING
 
-    def get_paid_date(self) -> Optional[datetime]:
-        if self.is_paid():
-            return self.payments[len(self.payments) - 1].date
-        return None
+def is_amount_counter_paid(counter: AmountCounter):
```
Comment (pskd73) [inline pyluca/amount_counter.py:45]: As you said, this logic should ideally not be in pyluca. Taking it out of AmountCounter is the first step towards it

**Pull request closed without merging**